tests/src/Functional/ServiceStatusTest.php

```diff
@@ -0,0 +1,191 @@
+    $this->assertSession()->pageTextContains('Test status body');
+
+    // See a status message on a landing page.
+    // Hide a status message on a landing page.
+  }
+
+  /**
+   * Test listings.
+   */
+  public function testServiceListingPages() {
+    $this->drupalLogin($this->adminUser);
+
+    // Create a landing page.
+    $landing_path = '/' . $this->randomMachineName(8);
+    $edit = [
+      'title[0][value]' => 'Test Service',
+      'body[0][summary]' => 'Test service summary',
+      'body[0][value]' => 'Test service body',
+      'status[value]' => 1,
+      'path[0][alias]' => $landing_path,
+    ];
+    $this->drupalPostForm('/node/add/localgov_services_landing', $edit, 'Save');
+
+    // Create some status updates.
+    for ($i = 1; $i <= 3; $i++) {
+      $edit = [
+        'title[0][value]' => 'Test Status ' . $i,
+        'body[0][value]' => 'Test service body ' . $i,
+        'field_service' => 1,
+        'field_service_status' => 'severe-impact',
+        'field_service_status_on_landing[value]' => 1,
+        'field_service_status_on_list[value]' => 1,
+        'status[value]' => 1,
+      ];
+      $this->drupalPostForm('/node/add/localgov_services_status', $edit, 'Save');
+    }
+
+    // Rebuild caches.
+    // See: https://github.com/localgovdrupal/localgov_services/issues/28
+    drupal_flush_all_caches();
+
+    // Check service status updates page.
+    $this->drupalGet($landing_path . '/status');
+    $this->assertSession()->statusCodeEquals(200);
+    $this->assertSession()->pageTextContains('Test Status 1');
+    $this->assertSession()->pageTextContains('Test Status 2');
+    $this->assertSession()->pageTextContains('Test Status 3');
+
+    // Check the service-status page.
+    $this->drupalGet('/service-status');
+    $this->assertSession()->statusCodeEquals(200);
+    $xpath = '//ul[@id="tabs"]/li/a';
+    /** @var \Behat\Mink\Element\NodeElement[] $results */
+    $results = $this->xpath($xpath);
+    $this->assertContains('Test Status 1', $results[0]->getText());
+    $this->assertContains('Test Status 2', $results[1]->getText());
+    $this->assertContains('Test Status 3', $results[2]->getText());
+
+    // Check sticky on top works.
+    $edit = [
+      'sticky[value]' => 1,
+    ];
+    $this->drupalPostForm('/node/4/edit', $edit, 'Save');
+    $this->drupalGet('/service-status');
+    $xpath = '//ul[@id="tabs"]/li/a';
+    /** @var \Behat\Mink\Element\NodeElement[] $results */
+    $results = $this->xpath($xpath);
+    $this->assertContains('Test Status 3', $results[2]->getText());
+    $this->assertContains('Test Status 1', $results[0]->getText());
+    $this->assertContains('Test Status 2', $results[1]->getText());
+
+    // Check unpublish.
+    $edit = [
+      'status[value]' => 0,
+    ];
+    $this->drupalPostForm('/node/3/edit', $edit, 'Save');
+
+    // Rebuild caches.
+    // See: https://github.com/localgovdrupal/localgov_services/issues/28
+    drupal_flush_all_caches();
```

Comment:
Is this not what `localgov_services_status_node_presave()` is supposed to do? Is it possibly not getting called in the test. Or is it not doing it's job?